src/plugins/cosmos/components/CosmosActionButtons/CosmosActionButtons.tsx

```diff
@@ -0,0 +1,29 @@
+import { Flex } from '@chakra-ui/layout'
+import { Button } from '@chakra-ui/react'
+import { StakingAction } from 'plugins/cosmos/components/modals/Staking/Staking'
+import { Text } from 'components/Text'
+import { useModal } from 'context/ModalProvider/ModalProvider'
+
+export const CosmosActionButtons = () => {
+  const { cosmosStaking } = useModal()
+
+  const handleUnstakeClick = () => {
+    cosmosStaking.open({ assetId: 'cosmoshub-4/slip44:118', action: StakingAction.Unstake })
```

Comment:
Since `cosmosStaking` is already opened, this effectively reinjects with new props.
IMO, this is the simplest and cleanest way to achieve staking/unstaking route change, instead of adding an additional nested memory router, with all the ramification it implies!